### PR TITLE
Create tests and scripts

### LINKED ISSUE #440: 🗑️Allow overwriting existing code in qawolf create
Delete a file if it already exists.

### PR DESCRIPTION
**Pull request by flaurida**
Resolves #440 #449

### CODE REVIEW

File: src/build-code/buildTemplate.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+export const buildScriptTemplate = (name: string): string => {
```
Comment (jperl) [inline src/build-code/buildTemplate.ts]: todo device/url and requires

**Comment by jperl** [comment]
Also please set up `npx qawolf create` cli command in this PR. For test just call runJest with the path. For script, create a runScript that calls `runCommand` with `node pathToScript`.

File: src/build-code/buildTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import { devices } from 'playwright';
+
+interface BuildTemplateOptions {
+  device?: string;
+  name: string;
+  url: string;
+}
+
+const QAWOLF_REQURE = 'const qawolf = require("qawolf");';
```
Comment (jperl) [inline src/build-code/buildTemplate.ts]: spelling

Comment (jperl) [reply src/build-code/buildTemplate.ts]: also I like REQUIRE_QAWOLF for some reason better

File: src/build-code/buildTemplate.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    ${buildNewContext(device)}
+  });
+
+  afterAll(() => browser.close());
+
+  test('${name}', async () => {
+    let page = await context.newPage();
+    await page.goto("${url}");
+
+    await qawolf.create();
```
Comment (jperl) [inline src/build-code/buildTemplate.ts]: no extra space

File: src/build-code/buildTemplate.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+export const buildScriptTemplate = ({
+  device,
+  name,
+  url,
+}: BuildTemplateOptions): string => {
+  const code = `${buildRequires(device)}
+
+const ${name} = async context => {
+  let page = await context.newPage();
+  await page.goto("${url}");
```
Comment (jperl) [inline src/build-code/buildTemplate.ts]: make spacing consistent, either add it or dont

File: src/build-code/buildTemplate.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+  beforeAll(async () => {
+    browser = await qawolf.launch();
+    ${buildNewContext(device)}
+  });
+
+  afterAll(() => browser.close());
+
+  test('${name}', async () => {
+    page = await context.newPage();
```
Comment (jperl) [inline src/build-code/buildTemplate.ts]: do the `page = await contex.newPage()` in the beforeAll

File: src/build-code/buildTemplate.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+export const buildScriptTemplate = ({
+  device,
+  name,
+  url,
+}: BuildTemplateOptions): string => {
+  const code = `${buildRequires(device)}
+
+const ${name} = async context => {
+  let page = await context.newPage();
```
Comment (jperl) [inline src/build-code/buildTemplate.ts]: do the `page = await context.newPage()` in the beforeAll

File: src/cli/cli.ts [hunk 7]
```diff
@@ -14,63 +14,64 @@ program.usage('<command> [options]').version(pkg.version);
...
+  .option('--repl', 'open a REPL when repl() is called')
+  .option('--webkit', 'run tests on webkit')
+  .description('run a test with Jest')
+  .allowUnknownOption(true)
+  .action((cmd = {}) => {
+    const args = omitArgs(process.argv.slice(3), [
+      '--all-browsers',
+      '--chromium',
+      '--firefox',
+      '-p',
```
Comment (jperl) [inline src/cli/cli.ts]: need to update -p -> -r

File: src/cli/cli.ts [hunk 8]
```diff
@@ -14,63 +14,64 @@ program.usage('<command> [options]').version(pkg.version);
...
+  .action(async (urlArgument, optionalName, { device, script }) => {
+    const url = parseUrl(urlArgument);
+    const name = optionalName || url.hostname!.replace(/\..*/g, '');
+
+    await saveTemplate({ device, name, url, script });
+  });
+
+program
+  .command('test')
+  .option('-r, --rootDir <rootDir>', 'directory where script will be saved')
```
Comment (jperl) [inline src/cli/cli.ts]: Change description back to 'path to test code'

File: src/cli/cli.ts [hunk 9]
```diff
@@ -14,63 +14,64 @@ program.usage('<command> [options]').version(pkg.version);
...
+      '--all-browsers',
+      '--chromium',
+      '--firefox',
+      '-p',
+      '--path',
+      '--repl',
+      '--webkit',
+    ]);
+
+    let browsers: BrowserName[] = [];
```
Comment (jperl) [inline src/cli/cli.ts]: make const

File: src/cli/saveTemplate.ts [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+): Promise<void> => {
+  try {
+    const { path, template } = buildTemplate(options);
+
+    if (!(await shouldSaveTemplate(path))) return;
+
+    await ensureFile(path);
+    return writeFile(path, template);
+  } catch (e) {
+    const templateType = options.script ? 'script' : 'test';
```
Comment (jperl) [inline src/cli/saveTemplate.ts]: why catch error? it should be thrown

Comment (flaurida) [reply src/cli/saveTemplate.ts]: if the error is thrown will log nicely in terminal?

Comment (jperl) [reply src/cli/saveTemplate.ts]: I don't think this makes it log nicer

File: .eslintrc.json [hunk 11]
```diff
@@ -19,7 +19,13 @@
     "prettier",
     "prettier/@typescript-eslint"
   ],
-  "ignorePatterns": ["*.d.ts", "docs/*", "sandbox/*"],
+  "ignorePatterns": [
+    "**/*.d.ts",
+    "**/*.js",
+    "docs/",
+    "sandbox/",
+    "node_modules/"
```
Comment (jperl) [inline .eslintrc.json]: abc :)

File: src/cli/saveTemplate.ts [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+      message: `"${path}" already exists, overwrite it?`,
+      name: 'overwrite',
+      type: 'confirm',
+    },
+  ]);
+
+  return overwrite;
+};
+
+// when running it pass the paths via environment variable
```
Comment (jperl) [inline src/cli/saveTemplate.ts]: you can remove this comment

File: test/build-code/__snapshots__/buildTemplate.test.ts.snap [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+    await myScript(context);
+    await browser.close();
+  })();
+}"
+`;
+
+exports[`buildTemplate includes device if specified 2`] = `
+"const { devices } = require(\\"playwright\\");
+  const qawolf = require(\\"qawolf\\");
+
```
Comment (jperl) [inline test/build-code/__snapshots__/buildTemplate.test.ts.snap]: i think we should not have a newline here

File: test/build-code/__snapshots__/buildTemplate.test.ts.snap [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+  })();
+}"
+`;
+
+exports[`buildTemplate includes device if specified 2`] = `
+"const { devices } = require(\\"playwright\\");
+  const qawolf = require(\\"qawolf\\");
+  
+  const device = devices[\\"iPad Mini\\"];
+
```
Comment (jperl) [inline test/build-code/__snapshots__/buildTemplate.test.ts.snap]: I think we should not have more than one newline here

File: test/build-code/__snapshots__/buildTemplate.test.ts.snap [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    const context = await browser.newContext();
+    await myScript(context);
+    await browser.close();
+  })();
+}"
+`;
+
+exports[`buildTemplate builds test template 1`] = `
+"const qawolf = require(\\"qawolf\\");
+
```
Comment (jperl) [inline test/build-code/__snapshots__/buildTemplate.test.ts.snap]: no newline

Comment (flaurida) [reply test/build-code/__snapshots__/buildTemplate.test.ts.snap]: i like one line after imports

**Pull request merged**